oap-server/analyzer/log-analyzer/src/main/java/org/apache/skywalking/oap/log/analyzer/dsl/spec/extractor/ExtractorSpec.java

```diff
@@ -290,13 +294,34 @@ public void slowSql(@DelegatesTo(SlowSqlSpec.class) final Closure<?> cl) {
             return;
         }
 
+        LocalDateTime localDateTime = Instant.ofEpochSecond(log.getTimestamp()).atZone(ZoneId.systemDefault()).toLocalDateTime();
+        String timeBucketForDB = DTF.format(localDateTime);
+        builder.setTimeBucket(Long.parseLong(timeBucketForDB));
+
         String entityId = serviceMeta.getEntityId();
         builder.prepare();
         DatabaseSlowStatement databaseSlowStatement = builder.toDatabaseSlowStatement();
         databaseSlowStatement.setDatabaseServiceId(entityId);
 
         sourceReceiver.receive(databaseSlowStatement);
         sourceReceiver.receive(serviceMeta);
+        // to service instance traffic
+        if (StringUtil.isNotEmpty(log.getServiceInstance())) {
+            ServiceInstanceUpdate instanceMeta = new ServiceInstanceUpdate();
+            instanceMeta.setServiceId(entityId);
+            instanceMeta.setName(namingControl.formatInstanceName(log.getServiceInstance()));
+            instanceMeta.setTimeBucket(timeBucket);
+            sourceReceiver.receive(instanceMeta);
+        }
+        // to endpoint traffic
+        if (StringUtil.isNotEmpty(log.getEndpoint())) {
+            EndpointMeta endpointMeta = new EndpointMeta();
+            endpointMeta.setServiceName(serviceName);
+            endpointMeta.setServiceNormal(true);
+            endpointMeta.setEndpoint(namingControl.formatEndpointName(serviceName, log.getEndpoint()));
+            endpointMeta.setTimeBucket(timeBucket);
+            sourceReceiver.receive(endpointMeta);
+        }
```

Comment:
I think we don't have a plan to lock slowSql for instance and endpoint level? Because the query doesn't support to do so, right?

Reply:
Got it.